src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp

```diff
@@ -0,0 +1,53 @@
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "GameCommands.h"
+#include "Types.hpp"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004B0B50
+    static uint32_t vehiclePassSignal(EntityId headId, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        try
+        {
+            Vehicles::Vehicle train(headId);
+            setPosition(train.veh2->position);
+
+            auto& head = train.head;
+            if (!head->canBeModified())
+            {
+                return FAILURE;
+            }
+
+            if (!(flags & Flags::apply))
+            {
+                return 0;
+            }
+
+            if (head->tileX == -1)
+            {
+                return 0;
+            }
+
+            head->var_5C = 2;
+            train.veh1->var_48 |= 1;
```

Comment:
Its hightime we convert this into an enum flags. Looks like (1<<0) is pass signal judging by all the other users.

Reply:
Any ideas what (1 << 1) and (1 << 2) mean? Tbh while I agree I'd prefer to handle this in another PR.

Reply:
okay make a follow on.

Reply:
Eh, it's less of a diff than I thought. Amended this PR instead.